### DISCUSSION TITLE
using foreignTable for sorting / filtering result

### DISCUSSION THREAD

**Discussion by jamorajj** [supabase/supabase#2343] [Questions] [answered]
Good day!

I've recently posted a follow up question in this [discussion](https://github.com/supabase/supabase/discussions/1169) but would like to expound on it. It involves the **foreignTable** property on .order() and .or()

case 1:
I'm trying to sort a list of results with the following call:
```
let { data, error } = await supabase
    .from('inventory')
    .select('*, product_id(name, sku, brand, category)')
    .eq('branch_id', id)
    .order('name', { foreignTable: 'product' });
```
wherein **inventory** table has a foreign key of **product_id** for **product** table.
Is there a way to sort the result by product name? The current call does not yield that result.

case 2:
I'm trying to do a search query to filter results and sort by name:
```
let { data, error } = await supabase
    .from('inventory')
    .select('*, product_id(name, sku, brand, category)')
    .eq('branch_id', id)
    .or(`name.ilike.%${term}%`, { foreignTable: 'product' })
    .order('name', { foreignTable: 'product' });
```
wherein **name** is a column of the foreign table **product**.
Result for this call does not filter the result and I'm getting the result as if the only filter being read is .eq().

Any insight or help into this would be very much appreciated!

**Comment by steve-chavez** [comment, marked as answer]
Hey @jamorajj,

Yes, right now the foreign table order doesn't affect the parent table(the one in the `from`). This is kind of related to https://github.com/supabase/postgrest-js/issues/197

A workaround for now would be creating a view of inventory that JOINs on products:

```sql
create or replace view inventory_products as
select
  i.*,
  p.name as product_name
from inventory i
join product p on p.id = i.product_id
```

```js
let { data, error } = await supabase
    .from('inventory_products')
    .select('*, product_id(name, sku, brand, category)')
    .eq('branch_id', id)
    .order('product_name');
```

---

Since your case is a many-to-one relationship, it might be possible to improve postgrest-js to be able to order in this way(order based on the parent row), but for now you'll have to do the above.

**Reply by jamorajj** [reply]
Hi @steve-chavez 

I was thinking of modifying the tables and to add a new table to achieve the result, but this did generate the result I was looking for. Thanks for your help on this!
